### PR TITLE
WIP: MM-9746 - display markdown svgs

### PR DESCRIPTION
**Pull request by RyanCommits**
- update svg package
- display svgs when detected

#### Summary
Integrated **react-native-svg** to display inline markdown SVG's.
Enter `[Mattermost](https://github.com/mattermost/mattermost-server)` to see svg displayed.

Helped and guidance appreciated because I've noticed multiple issues: 

1. Svg preview is nonexistent if you just enter the SVG url, unlike JPG's: `https://about.mattermost.com/wp-content/uploads/2017/08/Mattermost-Logo-Blue.svg`
2. If you tap on the displayed SVG, the full screen view will show a black screen
3. please review my function `isSvgLink`. It might not cover all the bases correctly.
4. ~~I couldn't use **ImageCacheManager's** image url as **react-native-svg** does not support a local system url, so I'm using `this.props.source`. My understanding of this may be incorrect.~~ (resolved)
5. My hardcoded `height` and `width` currently works for my android and iOS devices, but it may not work on others. A lot of UI testing might be required.

#1 and #2 are features unrelated to the focus of this ticket, in my opinion, as this ticket was made to address specifically Inline **Markdown** SVG's only. 

Would appreciate any suggestions on how I can improve this implementation.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-9746
Fixes https://github.com/mattermost/mattermost-server/issues/13219

#### Checklist

- [x] Has UI changes
#### Device Information
iPhone 6s, 13.3
iPhone XS Max Simulator, 13.3
Google Pixel 2 XL, Android 10

### CODE REVIEW

File: app/components/markdown/markdown_image/markdown_image.js [hunk 1]
```diff
@@ -268,6 +269,19 @@ export default class MarkdownImage extends React.Component {
...
+        } else if (isSvgLink(this.props.source)) {
+            image = (
+                <TouchableWithFeedback
+                    onLongPress={this.handleLinkLongPress}
+                    onPress={this.handlePreviewImage}
+                >
+                    <SvgUri
+                        width='100%'
+                        height='100'
+                        uri={this.props.source}
```
Comment (enahum) [inline app/components/markdown/markdown_image/markdown_image.js]: I would like to cache the svg's as we do with regular image files by using the `ImageCacheManager`, can this component load the SVG from a file in the filesystem?

File: app/components/markdown/markdown_image/markdown_image.js [hunk 2]
```diff
@@ -268,6 +269,19 @@ export default class MarkdownImage extends React.Component {
...
             }
+        } else if (isSvgLink(this.props.source)) {
+            image = (
+                <TouchableWithFeedback
+                    onLongPress={this.handleLinkLongPress}
+                    onPress={this.handlePreviewImage}
+                >
+                    <SvgUri
+                        width='100%'
+                        height='100'
```
Comment (enahum) [inline app/components/markdown/markdown_image/markdown_image.js]: this seem to be a random value, care to explain the reason to use `100`? what happens if the image height is lower?

Comment (RyanCommits) [reply app/components/markdown/markdown_image/markdown_image.js]: SVG images in this case scales to fit their container. 100 pixels just seemed like an appropriate value to get a good view of an image on mobile.

File: app/components/markdown/markdown_image/markdown_image.js (hunk 1 above)
Comment (RyanCommits) [reply app/components/markdown/markdown_image/markdown_image.js]: Changed this to load from file system. URI's needed to start with `file://` if it's a local url

Comment (RyanCommits) [reply app/components/markdown/markdown_image/markdown_image.js]: Mobile View (ios):
Screen Shot 2020-01-26 at 2 41 56 PM

Chrome Web view: 
Screen Shot 2020-01-26 at 2 42 03 PM

Note the lack of color on the Trustpilot svg. There are some SVG's that are rendered inconsistently. That might be an issue for a future ticket?

File: app/components/markdown/markdown_image/markdown_image.js (hunk 2 above)
Comment (RyanCommits) [reply app/components/markdown/markdown_image/markdown_image.js]: Mobile View (ios):
Screen Shot 2020-01-26 at 2 41 56 PM

Chrome Web view: 
Screen Shot 2020-01-26 at 2 42 03 PM

Note the lack of color on the Trustpilot svg. There are some SVG's that are rendered inconsistently. That might be an issue for a future ticket?

**Comment by RyanCommits** [comment]
@enahum How should we proceed? Not sure what @andrewbrown00 is referring to

**Comment by enahum** [comment]
@andrewbrown00 explain yourself dude... I'm on the same boat as @RyanCommits here.

For the regular images we are following the rules that were on the spec doc for images previews.

**Comment by andrewbrown00** [comment]
> @andrewbrown00 explain yourself dude... I'm on the same boat as @RyanCommits here.
> 
> For the regular images we are following the rules that were on the spec doc for images previews.

@enahum for sure...

What spec doc was referenced for this? I don't see it in the description or the JIRA ticket.

Initially, I was asking if you could point me to the current code/rules we have defined for standard images (non-svgs) to see what height we are using to align with that.

**Comment by enahum** [comment]
@andrewbrown00 I guess this one would be the reference? https://github.com/mattermost/mattermost-mobile/pull/3511

There is no spec doc for this I’m afraid. As far as what is done in the webapp for instance is unknown to me. That is why I’m adding you to the review on this PR so you can provide guidance on what needs to be done.

**Comment by andrewbrown00** [comment]
@enahum @RyanCommits let's use the following for:

**Mobile** 
- [Design Specs](https://projects.invisionapp.com/share/9XRRHMSBV6D#/screens/387495587), the implenetation of the `design specs` was done for non-SVGs in https://github.com/mattermost/mattermost-mobile/pull/3511 and the SVGs should follow the same rules as non-SVGs.

**WebApp**
I grabbed the _current rules_ from the inspector for reference. 0/5 if this is all that is needed; the key takeaways here are the `max-height` and `max-width`. All other styles/behavior should follow the current WebApp implementation.

```
.file-view--single .file__image .image-loaded img {
    align-self: center;
    cursor: pointer;
    max-height: 350px;
    position: relative;
    z-index: 2;
    border-radius: 4px;
    transition: all 0.1s linear 0s;
    overflow: hidden;
}

img {
    max-width: 100%;
}

img {
    vertical-align: middle;
}

img {
    border-width: 0px;
    border-style: initial;
    border-color: initial;
    border-image: initial;
}
```

Let me know if that clarifies things, or if you need anything else.

**Comment by RyanCommits** [comment]
@andrewbrown00 @enahum 
The problem with styling SVG's is that they don't come with a defined `height` and `width` since they automatically scale to whatever size we tell it to. We have to provide a container of a defined size for us to render a SVG in, so auto scaling won't work for us.

For example, if we define a container with `height: 350px` and `width: 100%`: 
**Vertical aspect ratio svg**: 

Screen Shot 2020-02-10 at 6 26 44 PM

**Horizontal aspect radio svg**:

Screen Shot 2020-02-10 at 6 27 29 PM

Both will take up the entire 350 pixels, (although the horizontal one is limited by its width and will try to maintain its aspect ratio). If we define a smaller container, both will scale down without losing aspect ratio. 

The question here is: **What is the height of the container we want to set for all SVG's to scale into?** From what I know, we cannot dynamically change the container size and customize it to an individual SVG.

Helpful article: https://css-tricks.com/scale-svg/

**Comment by enahum** [comment]
@RyanCommits @andrewbrown00  based on the above comment I can see different container sizes being implemented:
* Inline markdown images without text
*  Inline markdown images with text
* SVG as an attachment

**Comment by RyanCommits** [comment]
Now that I look deeper into this, I'm running into a slew of issues... I'll update this to WIP

**Comment by andrewbrown00** [comment]
Thanks for the context @RyanCommits, let me kick this around with the design team and @enahum and I'll follow up.

**Comment by RyanCommits** [comment]
I'm blocked by this issue: https://github.com/react-native-community/react-native-svg/issues/1202

I cannot figure out how to scale down a large svg image using `react-native-svg`

**Comment by jasonblais** [comment]
@migbot or @enahum wondering if you have any tips or guidance you can offer to @RyanCommits?

**Comment by hanzei** [comment]
@migbot @enahum Do you have any other ideas or guidance on how to continue?

**Comment by enahum** [comment]
This needs a UX pass for sure, I'm going to close the PR for now

**Pull request closed without merging**